project/SparkBuild.scala

```diff
@@ -765,16 +750,6 @@ object CopyDependencies {
 
 }
 
-object Java8TestSettings {
-  import BuildCommons._
-
-  lazy val settings = Seq(
-    javacJVMVersion := "1.8",
-    // Targeting Java 8 bytecode is only supported in Scala 2.11.4 and higher:
-    scalacJVMVersion := (if (System.getProperty("scala-2.10") == "true") "1.7" else "1.8")
```

Comment:
@srowen -- these builds grab the scala jar of the proper version depending on what axis is being built.  look in the dev/ directory at the code in there. 

i'm also CCing @JoshRosen as the build scripts have come a long way since i last poked at them.  :)